addons/l10n_de/models/datev.py

```diff
@@ -33,3 +33,23 @@ def _post(self, soft=True):
                             raise UserError(_('Account %s does not authorize to have tax %s specified on the line. \
                                 Change the tax used in this invoice or remove all taxes from the account') % (account_name, tax.name))
         return super()._post(soft)
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def _get_product_accounts(self):
+        """ As taxes with a different rate need a different income/expense account, we add this logic in case people only use
+         invoicing to not be blocked by the above constraint"""
+        result = super(ProductTemplate, self)._get_product_accounts()
+        company = result.get('income') and result['income'].company_id or False
+        if company and company.country_id.code == "DE":
+            taxes = self.taxes_id.filtered(lambda t: t.company_id == company)
+            if result['income'].tax_ids and taxes and taxes[0] not in result['income'].tax_ids:
+                result['income'] = self.env['account.account'].search([('internal_group', '=', 'income'), ('deprecated', '=', False),
+                                                               ('tax_ids', 'in', taxes.ids)], limit=1)
+            supplier_taxes = self.supplier_taxes_id.filtered(lambda t: t.company_id == company)
+            if result.get('expense') and result['expense'].tax_ids and supplier_taxes and supplier_taxes[0] not in result['expense'].tax_ids:
```

Comment:
If you have no income account but have an expense account, you will never reach that condition, as company will be False.